File: src/research/main/seed.clj
(ns research.main.seed
  (:require [research.domain.session :as session]
            [research.domain.task :as task]
            [research.storage.repository :as repo])
  (:import (java.util UUID)))

(defn seed
  "Create session."
  [data topic query processor language provider]
  (let [repo (repo/repo data)
        id (str (UUID/randomUUID))
        map {:id id
             :topic topic
             :tasks []
             :created (task/format (task/now))
             :query query
             :processor processor
             :language language
             :provider provider}
        value (session/session map)]
    (repo/append repo value)
    (println (str "Created session: " (subs id 0 8)))
    (println (str "Topic: " topic))
    (subs id 0 8)))

File: crates/research-cli/src/seed.rs
use std::path::Path;

use research_domain::session;
use research_domain::task;
use research_storage::repository::{self, Mutable};

/// Create session and return short identifier.
pub fn seed(
    data: &Path,
    topic: &str,
    query: &str,
    processor: &str,
    language: &str,
    provider: &str,
) -> String {
    let repo = repository::repo(data);
    let id = std::env::var("RESEARCH_SESSION_ID")
        .ok()
        .filter(|s| uuid::Uuid::parse_str(s).is_ok())
        .unwrap_or_else(|| uuid::Uuid::new_v4().to_string());
    let now = chrono::Local::now().naive_local();
    let stamp = task::format(&now);
    let value = serde_json::json!({
        "id": id,
        "topic": topic,
        "tasks": [],
        "created": stamp,
        "query": query,
        "processor": processor,
        "language": language,
        "provider": provider,
    });
    let item = session::session(&value);
    repo.append(item);
    println!("Created session: {}", &id[..8]);
    println!("Topic: {}", topic);
    id[..8].to_string()
}
